aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -80,6 +81,7 @@
   private AwsProperties awsProperties;
   private FileIO fileIO;
   private LockManager lockManager;
+  private CloseGroupHelper closeGroupHelper;
```

Comment:
Could this implement `CloseableGroup` instead of using a helper? That does basically the same thing only you can add closeables at any time.

Reply:
Just to ensure I understand the suggestion, do you mean to make this class implement `CloseableGroup` or move the logic from `CloseGroupHelper` to `CloseableGroup` and continue to declare a `private CloseableGroup closeableGroup = new CloseableGroup(){}` here? If the former, Ryan M and I had a discussion around this in https://github.com/apache/iceberg/pull/2878#discussion_r678098958 but I should be able to do the latter. Thanks!

Reply:
Oh, I see. You can't use `CloseableGroup` unless the base class extends it. That makes sense.

Reply:
Yeah, I like the idea to use `CloseableGroup` and merge the helper there. There is no reason for that class to be abstract -- that just prevents it from being used for composition rather than inheritance. We can also use `ExceptionUtil.castAndThrow` to fix up the exception handling in it and update `addCloseable` to accept `AutoCloseable`.

Reply:
Thank you! I have merged the new class into the existing `CloseableGroup` and make it no longer abstract. Please let me know if this was not your intended suggestion!